docs/migrate-to-ccm-with-csimigration.md

```diff
@@ -0,0 +1,198 @@
+
+Sometimes it can be hard to migrate an entire cluster to use the new `StorageClass`. This is where `CSIMigration` comes into play (see [CSIMigration design proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-migration.md)). With this enabled, calls to the in-tree volume APIs will call out to the CSI plugins. See [In-tree Storage Plugin to CSI Migration Design Doc](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190129-csi-migration.md) for more detail.
+
+## Before you begin
+
+Note: This guide works on a cluster created by `kubeadm`.
+
+Example `kubeadm.yaml`:
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+kubernetesVersion: 1.17.0-beta.1
+networking:
+  podSubnet: 10.96.0.0/16
+  serviceSubnet: 10.97.0.0/16
+controllerManager:
+  extraArgs:
+    cloud-provider: openstack
+    cloud-config: /etc/kubernetes/cloud.conf
+  extraVolumes:
+  - name: cloud
+    hostPath: /etc/kubernetes/cloud.conf
+    mountPath: /etc/kubernetes/cloud.conf
+    readOnly: true
+---
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: InitConfiguration
+localAPIEndpoint:
+  bindPort: 6443
+nodeRegistration:
+  kubeletExtraArgs:
+    cloud-provider: openstack
+    cloud-config: /etc/kubernetes/cloud.conf
+```
+
+## Migrate to openstack-cloud-controller-manager
+
+Remember... In the first step, we want to keep the in-tree volume API, but for all of the rest, we want to use `openstack-cloud-controller-manager`. So we can't just remove the `--cloud-provider` flag from `kube-controller-manager`. In fact, we must disable all cloud related controllers.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+kubernetesVersion: 1.17.0-beta.1
+networking:
+  podSubnet: 10.96.0.0/16
+  serviceSubnet: 10.97.0.0/16
+controllerManager:
+  extraArgs:
+    cloud-provider: openstack
+    cloud-config: /etc/kubernetes/cloud.conf
+    controllers: '*,bootstrapsigner,tokencleaner,-cloud-node-lifecycle,-route,-service'
+  extraVolumes:
+  - name: cloud
+    hostPath: /etc/kubernetes/cloud.conf
+    mountPath: /etc/kubernetes/cloud.conf
+    readOnly: true
+```
+
+And create new manifests for `kube-controller-manager`:
+
+```bash
+kubeadm init --config kubeadm-disable-controllers.yaml phase control-plane controller-manager
```

Comment:
Maybe make it more clear that the aforementioned yaml is the `kubeadm-disable-controllers.yaml`, for people who don't make the connection.